storage-serviceaccount-cmdline-sample/README.md

```diff
@@ -26,28 +26,28 @@ You may need to set your `JAVA_HOME`.
         git clone https://github.com/GoogleCloudPlatform/cloud-storage-docs-xml-api-examples.git
         cd cloud-storage-docs-xml-api-examples/storage-serviceaccount-cmdline-sample
 
-2. Update key.p12 file.
-3. Edit `StorageServiceAccountSample.java` to add your bucket and service account email.
+2. Update key.json file.
+3. Edit `StorageServiceAccountSample.java` to add your bucket.
 4. Compile and run
 
         mvn compile install
-        mvn -q exec:java
+        GOOGLE_APPLICATION_CREDENTIALS=key.json mvn -q exec:java
```

Comment:
The envvar GOOGLE_APPLICATION_CREDENTIALS is only need if you use the service account.
If you run it from the properly configured GCE instance, you don't need that envvar.

https://github.com/GoogleCloudPlatform/cloud-pubsub-logging-python
has an example README for authentication for various environment.